src/engine/enginebuffer.cpp

```diff
@@ -1004,24 +1004,27 @@ void EngineBuffer::processTrackLocked(
 
     m_scratching_old = is_scratching;
 
-    // Handle repeat mode
-    bool at_start = m_filepos_play <= 0;
-    at_end = m_filepos_play >= m_trackSamplesOld;
-
-    bool repeat_enabled = m_pRepeat->toBool();
+    // If we're repeating and crossed the track boundary, ReadAheadManager already
+    // wrapped around the playposition.
+    // To ensure quantize is respected we request a phase sync.
+    // TODO(ronso) This just restores previous repeat+quantize behaviour. I'm not
+    // sure whether that was actually desired or just a side effect of seeking.
+    // Ife it's really desired, should this be moved to looping control in order
+    // to set the sync'ed playposition right away and fill the wrap-around buffer
+    // with correct samples from the sync'ed loop in / track start position?
+    if (m_pRepeat->toBool() && m_pQuantize->toBool() &&
+            (m_filepos_play > playpos_old) == backwards) {
+        // TODO() The resulting seek is processed in the following callback
+        // That is to late
+        requestSyncPhase();
```

Comment:
requestSyncPhase() requires an other track playing to be in sync. Did you consider a solution where the repeat + quantize becomes a beat loop that the track stays in sync whith itself?

Reply:
yes, this is only supposed to sync phase when there's another track. I'm still not sure if that's a good idea, and for which use case it helps, considering users expecting beat and phase sync would set the loop manually, i.e. don't rely on phase sync to snap to the correct beat.

and no, this should not be an extension of the original beat loop fix.